parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -488,31 +488,39 @@ loop = do
-              mdType <- eval $ LoadTypeOfTerm mdValue
-              case mdType of
-                Nothing -> respond $ MetadataMissingType ppe r
-                Just ty -> do
-                  let steps =
-                        second (const . step $ Type.toReference ty)
-                        . first (Path.unabsolute .  resolveToAbsolute) <$> srcs
-                  let get = Branch.head <$> use root
-                  before <- get
-                  stepManyAt steps
-                  after <- get
-                  (ppe, outputDiff) <- diffHelper before after
-                  respondNumbered $ ShowDiffNamespace
-                     Path.absoluteEmpty Path.absoluteEmpty ppe outputDiff
-                where
-                step mdType b0 = let
-                  tmUpdates terms = foldl' go terms srcle
-                    where
-                    go terms src = op (src, mdType, mdValue) terms
-                  tyUpdates types = foldl' go types srclt
-                    where
-                    go types src = op (src, mdType, mdValue) types
-                  in over Branch.terms tmUpdates . over Branch.types tyUpdates $ b0
-            (_srcle, _srclt, mdValues) ->
-              respond $ MetadataAmbiguous ppe mdValues
+             let defnsAndTerms = Map.fromList $ srcs `zip` (toList . getHQ'Terms <$> srcs)
+             let noReferents = Map.keys $ Map.filterWithKey (\_ v -> null $ v) defnsAndTerms
+             let paths = (\(path', segment) -> Path.snoc' path' $ HQ'.toName segment) <$> noReferents
+             when (not . null $ noReferents) $ respond $ DefinitionsNotFound paths
+             unless (not. null $ noReferents) $ do
+                mdType <- eval $ LoadTypeOfTerm mdValue
+                case mdType of
+                  Nothing -> respond $ MetadataMissingType ppe r
+                  Just ty -> do
+                    let steps =
+                          second (const . step $ Type.toReference ty)
+                          . first (Path.unabsolute . resolveToAbsolute) <$> srcs
+                    let get = Branch.head <$> use root
+                    before <- get
+                    stepManyAt steps
+                    after <- get
+                    (ppe, outputDiff) <- diffHelper before after
+                    respondNumbered $ ShowDiffNamespace
+                       Path.absoluteEmpty Path.absoluteEmpty ppe outputDiff
+                  where
+                  step mdType b0 = let
+                    tmUpdates terms = foldl' go terms srcle
+                      where
+                      go terms src = op (src, mdType, mdValue) terms
+                    tyUpdates types = foldl' go types srclt
+                      where
+                      go types src = op (src, mdType, mdValue) types
+                    in over Branch.terms tmUpdates . over Branch.types tyUpdates $ b0
+            (_srcle, _srclt, []) ->
+              respond $ MetadataNotFound $ HQ.toName mdValue2
```

Comment:
Instead of passing `Maybe Name`, which would be much use if it's `Nothing`, try just passing `mdValue2` directly.

Reply:
In which case this becomes a use of `SearchTermsNotFound`, right?